test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1

```diff
@@ -0,0 +1,104 @@
+Describe "Test-Json" -Tags "CI" {
+    BeforeAll {
+        $validSchemaJson = @"
+            {
+            'description': 'A person',
+            'type': 'object',
+            'properties': {
+                'name': {'type': 'string'},
+                'hobbies': {
+                'type': 'array',
+                'items': {'type': 'string'}
+                }
+                }
+            }
+"@
+
+        $invalidSchemaJson = @"
+            {
+            'description',
+            'type': 'object',
+            'properties': {
+                'name': {'type': 'string'},
+                'hobbies': {
+                'type': 'array',
+                'items': {'type': 'string'}
+                }
+                }
+            }
+"@
+
+        $validJson = @"
+            {
+                'name': 'James',
+                'hobbies': ['.NET', 'Blogging', 'Reading', 'Xbox', 'LOLCATS']
+            }
+"@
+
+        $invalidTypeInJson = @"
+            {
+                'name': 123,
+                'hobbies': ['.NET', 'Blogging', 'Reading', 'Xbox', 'LOLCATS']
+            }
+"@
+
+        $invalidTypeInJson2 = @"
+            {
+                'name': 123,
+                'hobbies': [456, 'Blogging', 'Reading', 'Xbox', 'LOLCATS']
+            }
+"@
+
+        $invalidNodeInJson = @"
+            {
+                'name': 'James',
+                'hobbies': ['.NET', 'Blogging', 'Reading', 'Xbox', 'LOLCATS']
+                errorNode
+            }
+"@
+}
+
+    It "Json is valid" {
+        Test-Json -Json $validJson | Should Be $true
+    }
+
+    It "Json is valid aganist a valid schema" {
+        Test-Json -Json $validJson -Schema $validSchemaJson | Should Be $true
+    }
+
+    It "Json is invalid" {
+        Test-Json -Json $invalidNodeInJson -ErrorAction SilentlyContinue | Should Be $false
+    }
+
+    It "Json is invalid aganist a valid schema" {
+        Test-Json -Json $invalidTypeInJson2 -Schema $validSchemaJson -ErrorAction SilentlyContinue | Should Be $false
+        Test-Json -Json $invalidNodeInJson -Schema $validSchemaJson -ErrorAction SilentlyContinue | Should Be $false
+    }
+
+    It "Test-Json throw if a schema is invalid" {
+        { Test-Json -Json $validJson -Schema $invalidSchemaJson -ErrorAction Stop } | ShouldBeErrorId "InvalidJsonSchema,Microsoft.PowerShell.Commands.TestJsonCommand"
+    }
+
+    It "Test-Json write an error on invalid (type) Json aganist a valid schema" {
+        $errorVar = $null
+        Test-Json -Json $invalidTypeInJson -Schema $validSchemaJson -ErrorVariable errorVar -ErrorAction SilentlyContinue
+
+        $errorVar.FullyQualifiedErrorId | Should BeExactly "InvalidJsonAgainistSchema,Microsoft.PowerShell.Commands.TestJsonCommand"
+    }
+
+    It "Test-Json write an error on invalid (node) Json aganist a valid schema" {
```

Comment:
This one and the one above can be combined into single It with testcases.

Reply:
Done.